adapters/adyoulike/adyoulike.go

```diff
@@ -0,0 +1,159 @@
+package adyoulike
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+
+	"github.com/buger/jsonparser"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+)
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	return &AdYouLikeAdapter{
+		endpoint: config.Endpoint,
+	}, nil
+}
+
+type AdYouLikeAdapter struct {
+	endpoint string
+}
+
+func (adapter *AdYouLikeAdapter) MakeRequests(
+	openRTBRequest *openrtb.BidRequest,
+	reqInfo *adapters.ExtraRequestInfo,
+) (
+	requestsToBidder []*adapters.RequestData,
+	errs []error,
+) {
+	var err error
+	var tagID string
+
+	if len(openRTBRequest.Imp) > 0 {
```

Comment:
The length check isn't strictly necessary. This code will only be called if there is at least 1 impression for your adapter.

Reply:
Did you mean to only check for unmarshal errors in the very first element or the `[]Imp` array? (`true` branch of the `if` statement in line 36) I ask this because I don't see anything else is done with the unmarshaled `ttxExt`.  

If we only meant to check for unmarshal errors in the very first element, maybe we could ommit the `if` statement altogether because Prebid Server core performs validation on all `Imp.Ext`s and will make sure that a `"placement"` field is present because it was labeled as `required` in `static/bidder-params/adyoulike.json`
```
26   func (adapter *AdYouLikeAdapter) MakeRequests(
27       openRTBRequest *openrtb.BidRequest,
28       reqInfo *adapters.ExtraRequestInfo,
29   ) (
30       requestsToBidder []*adapters.RequestData,
31       errs []error,
32   ) {
33       var err error
34       var tagID string
35  
36 -     if len(openRTBRequest.Imp) > 0 {
37 -         var imp = &openRTBRequest.Imp[0]
38 -         var bidderExt adapters.ExtImpBidder
39 -         if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
40 -             errs = append(errs, newBadInputError(err.Error()))
41 -         }
42 -
43 -         var ttxExt openrtb_ext.ExtImpAdyoulike
44 -         if err := json.Unmarshal(bidderExt.Bidder, &ttxExt); err != nil {
45 -             errs = append(errs, newBadInputError(err.Error()))
46 -         }
47 -     }
48 -
49 -     if len(openRTBRequest.Imp) == 0 {
50 -         errs = append(errs, newBadInputError("No impression in the bid request"))
51 -     }
52  
53       reqCopy := *openRTBRequest
54       reqCopy.Imp = []openrtb.Imp{}
adapters/adyoulike/adyoulike.go
```
What are your thoughts?

Reply:
This is a good point. I missed that the `ttxExt` variable isn't being used. I agree this code doesn't seem necessary. PBS-Core will perform this validation and then I think you're good to use jsonparser to access the data. That library is quicker for accessing a single field. It skips validation, but other parts of PBS-Core have that covered.

Reply:
I see,  removed this part.